File: gatekeeper/src/main/java/co/blustor/identity/gatekeeper/events/DisconnectEvent.java
package co.blustor.identity.gatekeeper.events;

import android.support.annotation.NonNull;

import co.blustor.identity.gatekeeper.callbacks.DisconnectCallback;

import static co.blustor.identity.gatekeeper.BluetoothConstants.EventName;

public class DisconnectEvent implements Event {

    private DisconnectCallback mCallback;

    public DisconnectEvent(DisconnectCallback callback) {
        mCallback = callback;
    }

    @NonNull
    @Override
    public EventName getName() {
        return EventName.DISCONNECT;
    }

    public DisconnectCallback getCallback() {
        return mCallback;
    }
}

File: gatekeeper/src/main/java/co/blustor/identity/gatekeeper/events/DisconnectEvent.kt
package co.blustor.identity.gatekeeper.events

import co.blustor.identity.gatekeeper.BluetoothConstants.EventName
import co.blustor.identity.gatekeeper.callbacks.DisconnectCallback

class DisconnectEvent(val callback: DisconnectCallback) : Event {

    override val name = EventName.DISCONNECT
}
